platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -118,6 +121,17 @@ @implementation MGLMapSnapshotter {
     std::unique_ptr<mbgl::Actor<mbgl::MapSnapshotter::Callback>> _snapshotCallback;
 }
 
+- (void)dealloc {
+    if (_snapshotCallback) {
```

Comment:
It probably should check for `_completion` - but it's very 50/50, since these things are currently tied together. I suspect that since I added the `completion` property we might be able to remove `loading` (and make the completion block `nonnull`).